irods/manager/data_object_manager.py

```diff
@@ -161,8 +161,11 @@ def put(self, local_path, irods_path, return_data_object = False, num_threads =
                     for chunk in chunks(f, self.WRITE_BUFFER_SIZE):
                         o.write(chunk)
         if kw.ALL_KW in options:
-            options[kw.UPDATE_REPL_KW] = ''
-            self.replicate(obj, **options)
+            repl_options = options.copy()
+            repl_options[kw.UPDATE_REPL_KW] = ''
+            del repl_options[kw.REG_CHKSUM_KW]
```

Comment:
It may be helpful to leave a comment explaining why `REG_CHKSUM_KW` is not valid in this case. That is, `REG_CHKSUM_KW` registers a checksum for the replica created above and the checksum for the replica created here will be verified against that, which is incompatible with `REG_CHKSUM_KW` in the replication API. ...or whatever makes sense to you.